src/iOS/toga_iOS/widgets/scrollcontainer.py

```diff
@@ -0,0 +1,104 @@
+from rubicon.objc import CGSizeMake
+from toga_iOS.libs import (
+    NSLayoutAttributeBottom,
+    NSLayoutAttributeHeight,
+    NSLayoutAttributeLeading,
+    NSLayoutAttributeTop,
+    NSLayoutAttributeTrailing,
+    NSLayoutAttributeWidth,
+    NSLayoutConstraint,
+    NSLayoutRelationEqual,
+    UIScrollView
+)
+from toga_iOS.window import (
+    iOSViewport,
+    UIColor
+)
+from travertino.size import at_least
+from .base import Widget
+
+
+class ScrollContainer(Widget):
+    __current_content = None
+    __scroll_height_constraint = None
+    __scroll_width_constraint = None
+    __vertical_enabled = False
+    __horizontal_enabled = False
+    
+    def __update_content_size(self):
+        # We need a layout pass to figure out how big the scrollable area should be
+        self.__current_content.interface.refresh()
+        
+        content_width = 0
+        padding_horizontal = 0
+        content_height = 0
+        padding_vertical = 0
+        
+        if self.__horizontal_enabled:
+            content_width = self.__current_content.interface.layout.width
+            padding_horizontal = self.__current_content.interface.style.padding_left + self.__current_content.interface.style.padding_right
+        else:
+            content_width = self.native.frame.size.width
+        
+        if self.__vertical_enabled:
+            content_height = self.__current_content.interface.layout.height
+            padding_vertical = self.__current_content.interface.style.padding_top + self.__current_content.interface.style.padding_bottom
+            # pad the scrollview for the statusbar offset
+            padding_vertical = padding_vertical + self.__current_content.viewport.statusbar_height
+        else:
+            content_height = self.native.frame.size.height
+
+        self.native.setContentSize_(CGSizeMake(content_width + padding_horizontal,
+                                               content_height + padding_vertical))
+
+    def __constrain_to_scrollview(self, widget):
+        # The scrollview should know the content size as long as the
+        # view contained has an intrinsic size and the constraints are
+        # not ambiguous in any axis.
+        view = widget.native
+        horizontal = NSLayoutConstraint.constraintsWithVisualFormat_options_metrics_views_(
+            'H:|[view]|',
```

Comment:
I prefer avoiding the "metrics" API for constraints; I don't find Yet Another Syntax that easy to parse, and it's more work internally for Objective C to parse as well.

Reply:
Got it, yup looking at constraints.py we are using the more explicit API - I'll make the changes for this